newIDE/app/src/UI/SemiControlledTextField.js

```diff
@@ -121,7 +144,17 @@ export default class SemiControlledTextField extends React.Component<
           if (!commitOnBlur) onChange(newValue);
         }}
         onBlur={event => {
-          onChange(event.currentTarget.value);
+          let value = event.currentTarget.value;
+          if (this.props.calculValue) {
+            const rex = /(^\d).*[-+*()/.,].*(\d+$)|(^\d+$)|^((\d)|[-+*()/.,]).((\d)|[-+*()/.,])+$/gm;
+            const regexData = value.match(rex);
+
+            if (regexData) {
+              value = eval(regexData[0]);
+            }
```

Comment:
I would move the whole evaluating to https://github.com/4ian/GDevelop/blob/b6a5784968a8e794667b3f3f0f3b80b5695c07cd/newIDE/app/src/PropertiesEditor/index.js#L270